packages/pods-core/src/builtin/GithubPod.ts

```diff
@@ -0,0 +1,318 @@
+import { ImportPod, ImportPodConfig, ImportPodPlantOpts } from "../basev3";
+import { JSONSchemaType } from "ajv";
+import { PodUtils } from "../utils";
+import { graphql } from "@octokit/graphql";
+import _ from "lodash";
+import {
+  DendronError,
+  DVault,
+  NoteProps,
+  NoteUtils,
+  stringifyError,
+  Time,
+  DEngineClient,
+} from "@dendronhq/common-all";
+
+const ID = "dendron.github";
+
+type GithubImportPodCustomOpts = {
+  owner: string;
+  repository: string;
+  to?: string;
+  from?: string;
+  status: string;
+  token: string;
+  fname: string;
+};
+
+type GithubAPIOpts = GithubImportPodCustomOpts & {
+  created?: string;
+  afterCursor?: string;
+};
+
+type GithubImportPodConfig = ImportPodConfig & GithubImportPodCustomOpts;
+
+export type GithubImportPodPlantOpts = ImportPodPlantOpts;
+
+export class GithubImportPod extends ImportPod<GithubImportPodConfig> {
+  static id: string = ID;
+  static description: string = "import github issues";
+
+  get config(): JSONSchemaType<GithubImportPodConfig> {
+    return PodUtils.createImportConfig({
+      required: ["repository", "owner", "status", "token", "fname"],
+      properties: {
+        owner: {
+          type: "string",
+          description: "owner of the repository",
+        },
+        repository: {
+          description: "github repository to import from",
+          type: "string",
+        },
+        status: {
+          type: "string",
+          description: "status of issue open/closed",
+          enum: ["open", "closed"],
+        },
+        to: {
+          type: "string",
+          description: "import issues created before this date: YYYY-MM-DD",
+          format: "date",
+          default: Time.now().toISODate(),
+          nullable: true,
+        },
+        from: {
+          type: "string",
+          description: "import issues created after this date: YYYY-MM-DD",
+          format: "date",
+          nullable: true,
+        },
+        token: {
+          type: "string",
+          description: "github personal access token",
+        },
+        fname: {
+          type: "string",
+          description: "name of hierarchy to import into",
+        },
+      },
+    }) as JSONSchemaType<GithubImportPodConfig>;
+  }
+
+  /*
+   * method to fetch issues from github graphql api
+   */
+  getDataFromGithub = async (opts: Partial<GithubAPIOpts>) => {
+    let result;
+    const { owner, repository, status, created, afterCursor, token } = opts;
+    let queryVal = `repo:${owner}/${repository} is:issue is:${status}`;
+    if (!_.isUndefined(created)) {
+      queryVal = queryVal.concat(` ${created}`);
+    }
+
+    const query = `query search($val: String!, $after: String)
+    {search(type: ISSUE, first: 100, query: $val, after: $after) {
```

Comment:
nit: extract 100 to const with name PAGE_SIZE or something similiar (I assume this is the page size?)

Reply:
okay. Page size is actually 30. 100 is the max no of data that can be fetched at a time.

Reply:
o I see - does 100 represent the max number of issues to pull in? like `TOP 100` in a SQL query? 

No need to amend the page size to 100 from 30 in that case.

Reply:
Yep- the first 100 from the stack.